hbase-server/src/test/java/org/apache/hadoop/hbase/security/access/TestReadOnlyController.java

```diff
@@ -0,0 +1,105 @@
+import org.apache.hadoop.hbase.master.HMaster;
+import org.apache.hadoop.hbase.regionserver.HRegion;
+import org.apache.hadoop.hbase.regionserver.RegionCoprocessorHost;
+import org.apache.hadoop.hbase.testclassification.LargeTests;
+import org.apache.hadoop.hbase.testclassification.SecurityTests;
+import org.apache.hadoop.hbase.util.Bytes;
+import org.junit.AfterClass;
+import org.junit.BeforeClass;
+import org.junit.ClassRule;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.rules.ExpectedException;
+import org.junit.rules.TestName;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import java.io.IOException;
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertTrue;
+import static org.junit.Assert.fail;
+
+@Category({ SecurityTests.class, LargeTests.class })
+@SuppressWarnings("deprecation")
+public class TestReadOnlyController {
+
+  @ClassRule
+  public static final HBaseClassTestRule CLASS_RULE =
+    HBaseClassTestRule.forClass(TestReadOnlyController.class);
+
+  private static final Logger LOG = LoggerFactory.getLogger(TestAccessController.class);
+  private static final HBaseTestingUtil TEST_UTIL = new HBaseTestingUtil();
+  private static TableName TEST_TABLE = TableName.valueOf("readonlytesttable");
+  private static byte[] TEST_FAMILY = Bytes.toBytes("readonlytablecolfam");
+  private static Configuration conf;
+  private static Connection connection;
+
+  private static RegionServerCoprocessorEnvironment RSCP_ENV;
+
+  private static Table TestTable;
+  @Rule
+  public TestName name = new TestName();
+
+  @Rule
+  public ExpectedException exception = ExpectedException.none();
+  @BeforeClass
+  public static void beforeClass() throws Exception {
+    conf = TEST_UTIL.getConfiguration();
+    // Only try once so that if there is failure in connection then test should fail faster
+    conf.setInt("hbase.ipc.client.connect.max.retries", 1);
+    // Shorter session timeout is added so that in case failures test should not take more time
+    conf.setInt(HConstants.ZK_SESSION_TIMEOUT, 1000);
+    // Enable ReadOnly mode for the cluster
+    conf.setBoolean(HConstants.HBASE_GLOBAL_READONLY_ENABLED_KEY, true);
+    // Add the ReadOnlyController coprocessor to interrupt any write operation
+    conf.set(CoprocessorHost.REGION_COPROCESSOR_CONF_KEY, ReadOnlyController.class.getName());
+    // Start the test cluster
+    TEST_UTIL.startMiniCluster(2);
+    // Get connection to the HBase
+    connection = ConnectionFactory.createConnection(conf);
+  }
+
+  @AfterClass
+  public static void afterClass() throws Exception {
+    // Cleanup for Connection and HBase cluster
+    connection.close();
+    TEST_UTIL.shutdownMiniCluster();
+  }
+
+  @Test
+  public void testReadOnlyEnabled() {
+    Configuration c = TEST_UTIL.getConfiguration();
+    assertTrue(c.getBoolean(HConstants.HBASE_GLOBAL_READONLY_ENABLED_KEY, false));
+  }
```

Comment:
You don't need this test, because you don't want to validate whether HBase configuration works or not.